src/api/wix-api-context-provider.tsx

```diff
@@ -106,7 +109,61 @@ function getWixApi(wixClient: ReturnType<typeof getWixClient>) {
             // return { success: true, url: redirectSession?.fullUrl };
             return { success: true, url: '' };
         },
-        getMyProfile: async () => await wixClient.members.getCurrentMember(),
+        getMyProfile: async () => await wixClient.members.getCurrentMember({}),
+        initiateLogin: async () => {
+            const loginRequestData = wixClient.auth.generateOAuthData(
+                // Add your redirect URI here after adding to the "Allowed redirect domains in settings"
+                "http://localhost:5173",
+                window.location.href
+            )
+            localStorage.setItem('oauthData', JSON.stringify(loginRequestData));
+            const { authUrl } = await wixClient.auth.getAuthUrl(loginRequestData);
+            window.location.href = authUrl;
+        },
+        fetchUserAuthData: async () => {
+            userAuthPromise = new Promise((res) => {
+                const wixTokens = JSON.parse(localStorage.getItem('wixTokens') || 'null');
+                if (wixTokens) {
+                    wixClient.auth.setTokens(wixTokens);
+                    wixApi.getMyProfile().then((profile) => {
+                        res({ user: profile });
+                    });
+                } else if (wixClient.auth.loggedIn()) {
+                    return wixApi.getMyProfile().then((profile) => {
```

Comment:
Why return this promise? in the above appearance of `wixApi.getMyPofile()` it's not being returned. Is this difference on purpose?